### DISCUSSION TITLE
index_not_found_exception

### DISCUSSION THREAD

**Discussion by BritneyStars** [apache/skywalking#8343] [Q&A]
I update the oap server service from version 8.5 to 8.9.1 , and I delete the all indices and all templates . but I found some indexes cannot be rebuilt normally, such as sw_profile_task and sw_network_address_alias. I restarting the oap service some times, but these indices sre still missing. when I again use the the 8.5.0 oap service , these indice will be created.

**Comment by BritneyStars** [comment]
// StorageEsInstaller
boolean shouldUpdateTemplate = !esClient.isExistsTemplate(tableName);
if (shouldUpdateTemplate) {
  // ...
}
I debuged the code and found the shouldUpdateTemplate is always false . But when I use oap server 8.5.0 debug, the shouldUpdateTemplate  is true . the two tests done when the es7.8 indices and telmplates are deleted

**Reply by wu-sheng** [reply]
This usually means, template is still there.

**Reply by BritneyStars** [reply]
After I delete the all templates, I check out that is deleted, and not found using es query template grammar. It is confused.

**Reply by wu-sheng** [reply]
The e2e is using the new and clean ElasticSearch every time. I believe the creating mechanism is working.

**Reply by BritneyStars** [reply]
I find the reason. I delete the all templates, but do not delete the all index templates. Thanks!
